Modules/admin/admin_controller.php

```diff
@@ -122,13 +142,34 @@ function read_file($file, $lines) {
                     @fclose($fh);
                 }
                 
-                if ($route->subaction == 'getupdatelog') { 
+                if ($route->subaction == 'getupdatelog' && $session['admin']) { 
                     $route->format = "text";
                     ob_start();
                     passthru("cat /home/pi/data/emonpiupdate.log");
                     $result = trim(ob_get_clean());
                 }
                 
+                if ($route->subaction == 'downloadupdatelog' && $session['admin'])
+                {
+                    header("Content-Type: application/octet-stream");
+                    header("Content-Transfer-Encoding: Binary"); 
+                    header("Content-disposition: attachment; filename=emonpiupdate.log");
+                    header("Pragma: no-cache"); 
+                    header("Expires: 0");
+                    flush();
+                    if (file_exists("/home/pi/data/emonpiupdate.log"))
```

Comment:
the path for `emonpiupdate.log` is defined on line 133

https://github.com/joyrider3774/emoncms/blob/ebe7dc607da65ba2fcb5233bc12d8877f2176656/Modules/admin/admin_controller.php#L133

It would be best to reference this variable `$update_logfile` here rather than hardcode the abs path in another location